docs/user-interface/env_oauth_config.md

```diff
@@ -0,0 +1,46 @@
+---
+title: InstructLab UI .env & OAuth config.
+description: Setting up github OAuth and .env file
+logo: images/ilab_dog.png
+---
+
+This chapter is dedicated to the settings for running the User Interface locally.
+
+## .env
+
+```bash
+IL_UI_ADMIN_USERNAME=admin
+IL_UI_ADMIN_PASSWORD=password
+IL_UI_DEPLOYMENT=dev
+
+OAUTH_GITHUB_ID=<OAUTH_APP_ID>
+OAUTH_GITHUB_SECRET=<OAUTH_APP_SECRET>
+
+NEXTAUTH_SECRET=your_super_secret_random_string
+NEXTAUTH_URL=http://localhost:3000
+
+IL_GRANITE_API=<GRANITE_HOST>
+IL_GRANITE_MODEL_NAME=<GRANITE_MODEL_NAME>
+IL_MERLINITE_API=<MERLINITE_HOST>
+IL_MERLINITE_MODEL_NAME=<MERLINITE_MODEL_NAME>
+
+GITHUB_TOKEN=<TOKEN FOR OAUTH INSTRUCTLAB MEMBER LOOKUP>
+TAXONOMY_DOCUMENTS_REPO=github.com/instructlab-public/taxonomy-knowledge-docs
+NEXT_PUBLIC_AUTHENTICATION_ORG=<AUTHENTICATION_ORG>
+NEXT_PUBLIC_TAXONOMY_REPO_OWNER=<GITHUB_ACCOUNT>
+NEXT_PUBLIC_TAXONOMY_REPO=<REPO_NAME>
+```
+
+## OAuth 
+
+To set up the github OAuth settings for the UI, please refer to the steps below
+
+`Github settings` -> `Developer settings` -> `OAuth Apps` -> `Create New OAuth App` 
+
+Fill in the fields as shown below:
+
+![UI OAuth Details](../images/user-interface/ui_oauth_details.png)
+
+Set the `Homepage URL` and the `Authorization Callback URL` to `http://localhost:3000`.
+
+Update the .env files with the new ID and secret ID generated, `OAUTH_GITHUB_ID` = Client ID, `OAUTH_GITHUB_SECRET` = Client Secret
```

Comment:
Creation of the OAuth App will create the client_id, that is set to OAUTH_GITHUB_ID, but it doesn't automatically create the secret. Once the app is created, on the main page it provides an option to create a secret. Press the Create Secret button and it will generate a secret for you. Copy the secret and set it to OAUTH_GITHUB_SECRET value.

Reply:
If user don't want to setup the OAuth in his own org, we would recommend users to reach out to UI maintainers  over the #ui slack channel and they will be happy to share the details of the dev OAuth app setup for the instructlab-public org for test-taxonomy.